src/test/java/com/hivemq/cli/commands/swarm/run/SwarmRunStopCommandTest.java

```diff
@@ -0,0 +1,131 @@
+        errorTransformer = mock(SwarmApiErrorTransformer.class);
+
+        apiClient = mock(ApiClient.class);
+        when(runsApi.getApiClient()).thenReturn(apiClient);
+
+        final Error error = mock(Error.class);
+        when(errorTransformer.transformError(any())).thenReturn(error);
+    }
+
+    @Test
+    void invalidUrl_error() throws ApiException {
+        final SwarmRunStopCommand invalid = new SwarmRunStopCommand("invalid", 1, runsApi, commanderApi, errorTransformer, out);
+        assertEquals(-1, invalid.call());
+        verify(runsApi, times(0)).getRun(any());
+        verify(apiClient, times(0)).setBasePath(anyString());
+    }
+
+    @Test
+    void runIdGiven_exception_error() throws ApiException {
+        final SwarmRunStopCommand invalid = new SwarmRunStopCommand("http://localhost:8080", 1, runsApi, commanderApi, errorTransformer, out);
+
+        when(runsApi.stopRun(any(), any())).thenThrow(mock(ApiException.class));
+
+        assertEquals(-1, invalid.call());
+        verify(runsApi).stopRun(eq("1"), any());
+        verify(apiClient).setBasePath("http://localhost:8080");
+        verify(errorTransformer).transformError(any());
+    }
+
+    @Test
+    void runIdGiven_noException_success() throws ApiException {
+        final SwarmRunStopCommand invalid = new SwarmRunStopCommand("http://localhost:8080", 1, runsApi, commanderApi, errorTransformer, out);
+
+        assertEquals(0, invalid.call());
+        verify(runsApi).stopRun(eq("1"), any());
+        verify(apiClient).setBasePath("http://localhost:8080");
+        verify(errorTransformer, times(0)).transformError(any());
+    }
+
+
+    @Test
+    void noRunIdGiven_exception_error() throws ApiException {
+        final SwarmRunStopCommand invalid = new SwarmRunStopCommand("http://localhost:8080", null, runsApi, commanderApi, errorTransformer, out);
+
+        when(commanderApi.getCommanderStatus()).thenThrow(mock(ApiException.class));
+
+        assertEquals(-1, invalid.call());
+        verify(runsApi, times(0)).stopRun(eq("1"), any());
+        verify(apiClient).setBasePath("http://localhost:8080");
+        verify(errorTransformer).transformError(any());
+    }
+
+    @Test
+    void noRunIdGiven_noRunInProgress_success() throws ApiException {
+        final SwarmRunStopCommand invalid = new SwarmRunStopCommand("http://localhost:8080", null, runsApi, commanderApi, errorTransformer, out);
+
+        final CommanderStateResponse commanderStateResponse = mock(CommanderStateResponse.class);
+        when(commanderStateResponse.getRunId()).thenReturn(null);
+        when(commanderApi.getCommanderStatus()).thenReturn(commanderStateResponse);
+
+        assertEquals(0, invalid.call());
+        verify(runsApi, times(0)).stopRun(eq("1"), any());
+        verify(apiClient).setBasePath("http://localhost:8080");
+        verify(errorTransformer, times(0)).transformError(any());
+
+    }
+
+    @Test
+    void noRunIdGiven_runInProgress_success() throws ApiException {
```

Comment:
why success? dont get the meaning here

Reply:
The command should return with exit code 0. Stopping the current run, when none is in progress should not be an error or give further explanation. 

"No news is good news"